nixos/modules/system/boot/loader/grub/grub.nix

```diff
@@ -75,15 +74,15 @@ let
   bootDeviceCounters = fold (device: attr: attr // { "${device}" = (attr."${device}" or 0) + 1; }) {}
     (concatMap (args: args.devices) cfg.mirroredBoots);
 
-  convertedFont = (pkgs.runCommand "grub-font-converted.pf2" {}
+  convertedFont = (targetPkgs.runCommand "grub-font-converted.pf2" {}
```

Comment:
convertedFont and defaultSplash shouldn't need targetPkgs - they should be the same regardless of architecture.